core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
Maybe it's not really possible to have the item (param, header) name derived from the request reader. We should probably roll back to the approach when we always specify the validated item. I thinking about the following syntax:

```scala
RequiredParam("a") suchThat ("a", "is positive") { _ > 0 }
```

Where `suchThat` is a math term `:` http://mathworld.wolfram.com/SuchThat.html.

We still may have `withFiler` implemented in terms of `suchThat`:

```scala
def withFilter(...) => suchThat("request", "is valid")(p)
```

@rpless, @jenshalm what do you think about this?